source/PlayState.hx

```diff
@@ -62,16 +73,18 @@ class PlayState extends FlxState
 	{
 		super.update(elapsed);
 
-		// Uncomment to collide hero against all wall objects.
 		FlxG.collide(hero, walls);
 
-		// Uncomment to trigger custom logic when a player overlaps with a fireball.
 		FlxG.overlap(hero, fireballs, resolveHeroFireballOverlap);
 
-		// Wrap various objects if gone offscreen.
+		FlxG.overlap(hero, goal, heroGoalOverlap);
+
 		screenWrapObject(hero);
 		for (fireball in fireballs) {
 			screenWrapObject(fireball);
+
+		ifEnding();
+		ifWon();
```

Comment:
I don't think you meant to put these function calls inside your fireball object-wrapping loop. As-is, they get called way more often than they need to.